File: Java/src/session2/td/agregation/NaturelInductif.java
package session2.td.agregation;

public interface NaturelInductif extends EtatNaturelPur {

    EtatNaturelPur creerZero();
    EtatNaturelPur creerSuccesseur(EtatNaturelPur e);
    EtatNaturelPur creerNatAvecValeur(int n);

}

File: TypeScript/src/session2.td/agregation/NaturelInductif.ts
import {EtatNaturelPur} from './etatnaturelpur';

export interface NaturelInductif extends EtatNaturelPur {

    creerZero() : EtatNaturelPur;
    creerSuccesseur(e : EtatNaturelPur) : EtatNaturelPur;
    creerNatAvecValeur(n : number) : EtatNaturelPur;

}
